server/resp/src/main/java/org/infinispan/server/resp/commands/sortedset/internal/POP.java

```diff
@@ -0,0 +1,61 @@
+package org.infinispan.server.resp.commands.sortedset.internal;
+
+import io.netty.channel.ChannelHandlerContext;
+import org.infinispan.multimap.impl.EmbeddedMultimapSortedSetCache;
+import org.infinispan.server.resp.Consumers;
+import org.infinispan.server.resp.Resp3Handler;
+import org.infinispan.server.resp.RespCommand;
+import org.infinispan.server.resp.RespErrorUtil;
+import org.infinispan.server.resp.RespRequestHandler;
+import org.infinispan.server.resp.commands.ArgumentUtils;
+import org.infinispan.server.resp.commands.Resp3Command;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * Common implementation for ZPOP commands
+ */
+public abstract class POP extends RespCommand implements Resp3Command {
+   private final boolean min;
+   public POP(boolean min) {
+      super(-2, 1, 1, 1);
+      this.min = min;
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
+                                                      ChannelHandlerContext ctx,
+                                                      List<byte[]> arguments) {
+
+      byte[] name = arguments.get(0);
+      EmbeddedMultimapSortedSetCache<byte[], byte[]> sortedSetCache = handler.getSortedSeMultimap();
+
+      long count = 1;
+      if (arguments.size() > 1) {
+         try {
+            count = ArgumentUtils.toLong(arguments.get(1));
+            if (count < 0) {
```

Comment:
Finally we don't have to iterate backwards with negative values :laughing:

Reply:
no in pop xD